controller/scheduler-v2/scheduler.go

```diff
@@ -0,0 +1,292 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/gopkg.in/inconshreveable/log15.v2"
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/controller/utils"
+	"github.com/flynn/flynn/host/types"
+)
+
+type Scheduler struct {
+	utils.ControllerClient
+	utils.ClusterClient
+	log        log15.Logger
+	formations *Formations
+
+	jobs    *jobMap
```

Comment:
This is mostly modeled based on the existing scheduler, since when I started I didn't have a good enough mental model to understand what was going on. I think it's now in a place where I could clean it up.